### DISCUSSION TITLE
ExpiredTargetSdkVersion error when building for Amazon App Store

### DISCUSSION THREAD

**Discussion by Michoko92** [airsdk/Adobe-Runtime-Support#1975] [Q&A] [answered]
Hi,

I have a pretty urgent issue, while trying to generate an Amazon App Store build with the latest AIR versions (both 821 and 889 do the same thing).

In my application.xml, I have targetSDKVersion=12, as I want to support older Fire devices. But I then get the following error, and the build fails:

Error: Google Play requires that apps target API level 29 or higher.
 [ExpiredTargetSdkVersion]
        targetSdkVersion 12

So I tried to change targetSDKVersion to 29, but then I get the following error when I try to intall the APK on my old 3rd gen Fire tablet: INSTALL_FAILED_OLDER_SDK . It seems those devices use an old Android system (it is Fire OS 4.5.5.3).

I don't know what to do. I hope there is still a way to support older Amazon devices. (I don't think I can use the old building method, as I then get "error: Found tag id where item is expected" errors.)

Thank you very much for your help!

**Comment by FliplineStudios** [comment, marked as answer]
Have you tried including a lower minSdkVersion as well, to see if that helps?  In one of our recent Amazon builds we had this (both targetSdkVersion and minSdkVersion declared within the same tag):

`<uses-sdk android:minSdkVersion="19" android:targetSdkVersion="30" />`

We used 19 as a minimum since our ANEs required that, and I believe that's the basis for Fire OS 4.5 as well, so that might be worth a try!

**Comment by ajwfrost** [comment]
Hi

(Okay so @FliplineStudios has already made this same suggestion (thanks!) whilst I was still writing the below .. but I'll post it anyway!)

Couldn't you use the "minimum SDK" to set this to the older devices, and "target SDK" to be more recent per the requirements for the Play store?
https://stackoverflow.com/questions/24510219/what-is-the-difference-between-min-sdk-version-target-sdk-version-vs-compile-sd

Although, note that we have a minimum SDK level now of 16 (Android 4.1) because of the changes required for the new Scout app; and for any 64-bit build, the minimum is 21.

thanks

**Comment by Michoko92** [comment]
Thank you guys for your very fast reply, you're awesome! :)

Unfortunately, I already had a minSdkVersion set to 9 (very low, I know). I set it to 17, but still no luck. I still get the "INSTALL_FAILED_OLDER_SDK " error. It seems this error occurs when you try to install an APK using a recent sdk target onto a device with an old system. It has been working fine for years, but now the Android builder prevents using API level below 29, which could actually be an issue for many Amazon devices...

Do you have other suggestions, please? :)

**Reply by ajwfrost** [reply]
Are you able to inspect your APK within Android Studio, and see what the actual minimum version is? Plus, how are you building -> just for an APK, using target armv7? or using armv8 or a bundle?

I'm wondering whether we've got an issue with the bundle code always using 21, rather than the lower number, because the bundles include all the formats... but when creating an APK file with the new mechanism, it's using the same internal code-paths to prepare the project folder as the bundle code uses....

thanks

**Comment by Michoko92** [comment]
It finally worked!

I really don't know why it didn't when I had two separate lines in my manifest, one for minSdk, and the other for TargetSdk (like this):

```
<uses-sdk android:targetSdkVersion="30"/>
<uses-sdk android:minSdkVersion="19"/>
```


 I copied and pasted @FliplineStudios line instead, and it magically worked!

`<uses-sdk android:minSdkVersion="19" android:targetSdkVersion="30" />`

Thank you so much for your help, you saved my life! Cheers! :)

**Reply by ajwfrost** [reply]
Great! that's probably caused by the way ADT is parsing the XML, worth noting that it only works if they're in the same tag..!

thanks

**Reply by FliplineStudios** [reply]
Awesome, glad it worked!  Yeah that's a quirk with the targetSdkVersion/minSdkVersion, they both need to be included in the same tag for it to recognize both values.
